presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryDataFragment.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.memory;
+
+import com.facebook.presto.spi.HostAddress;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.json.JsonCodec;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.util.Objects.requireNonNull;
+
+public class MemoryDataFragment
+{
+    public static final JsonCodec<MemoryDataFragment> MEMORY_DATA_FRAGMENT_CODEC = jsonCodec(MemoryDataFragment.class);
+
+    private final HostAddress hostAddress;
+    private final long rows;
+
+    @JsonCreator
+    public MemoryDataFragment(
+            @JsonProperty("hostAddress") HostAddress hostAddress,
+            @JsonProperty("rows") long rows)
+    {
+        this.hostAddress = requireNonNull(hostAddress, "hostAddress is null");
+        this.rows = rows;
```

Comment:
Can you checkArgument that rows is positive? Want to make sure that if somebody implements deletes they will have to deal with that.